runtime.md

```diff
@@ -15,3 +15,27 @@ Runs a process in a container. Can be invoked several times.
 Not sure we need that from oc cli. Process is killed from the outside.
 
 This event needs to be captured by oc to run onstop event handlers.
+
+## Hooks
+Hooks allows one to run code before/after various lifecycle events of the container. The state of the container is passed to the hooks over a well-known fd, so the hooks could get the information they need to do their work.
+
+Hooks are placed under a well known directories under which they are search for and executed. The list of directories that will be looked up are:
+
+```
+    /run/ocf/hooks
+    /usr/lib/ocf/hooks
+```
+
+### Pre-start
+The pre-start hook is called after the container process is started but before the user supplied command is execed. It provides an opportunity to customize the container. In Linux, for e.g., the network namespace could be configured in this hook.
+
+### Post-stop
+The post-stop hook is called after the container process is stopped. Cleanup or debugging could be performed in such a hook.
+
+```
+    "hooks" : {
+        "pre-start": ["pre-start-hook", "arg1", "arg2"],
```

Comment:
This should be an array so that you can specify multiple binaries for a single hook.

```json
"pre-start": [
    ["ps", "aux"],
    ["ip", "a"]
]
```